docs/02-about-us/guilds.md

```diff
@@ -0,0 +1,44 @@
+# Agile Guilds
```

Comment:
There is some discussion on renaming these Practice Areas

Reply:
@nerdstein I have heard of that, but given that that is a cosmetic change, I would say that is not a merge blocker.

Reply:
Practice Areas are not what this process describes. I do not consider what we're doing with the UX Practice Area, to be a Guild—for instance.

Reply:
I like simple language, and i think that "practice area" is simpler than "guild", and would be more easily understood by new hires, or people looking in from outside. Also: guild is historically a much more exclusive term, whereas i think our conception of these groups is much more fluid, and "cross-_practice_", despite an emphasis on craft.

Reply:
Sure... but this document specifically talks about "Guilds," which are a separate concept from a "Practice." "Practices" aren't beholden to rules across disciplines, because that makes no sense—whereas via the cross-silo purpose with Guilds, they need that uniform structure it sounds like, for optimal value.

Reply:
@ninavizz Ok, so it sounds like naming this process "practice area" would be a semantically incorrect correction :) . But, @kevwalsh likes it. I think this process does not have to be unique to a practice area (Ex. we are creating a DKAN guild even though this is not a practice area per se), that is why I think guild is a better (more abstract) name. But given that we are not stepping on your conception of what a "practice area" is @ninavizz, and that we can name this process whatever we want later on, can we go with guild for now?

Reply:
@ninavizz - right now, we don't have anything consistent for practice areas to meet regularly or evolve. The whole concept of a guild was to embody this idea. But, the lines between a guild and a practice area are very blurry. Since this is a new concept, i would almost recommend we start very basically with our definition of a Practice Area and evolve it later if we get feedback on how the Guild concept more specifically applies.

Reply:
@fmizzell - i don't think we would create a DKAN guild. We could create an Open Data guild because that is a practice

Reply:
ha, it seems like we keep on swaying to both sides, should we talk about this in person if we can?

Reply:
@fmizzell i don't see how "guild" could be more abstract.  historically, it's a less abstract concept, and based more closely on a specific craft, to the exclusion of those who do not practice the craft.

Reply:
My only wish is to call a spade a spade–and a "Practice Area" is not what this thing is describing. Guild is very, very specific to what these governance docs outline; really, this is not that. Please to no. Sorry, I gotta go focus on DMV site stuff, but just wanted to pipe-up that... like, the UX Practice stuff I don't see following any of these processes. That needs to be ok, because the prodev of practitioners and the value offered to engagements, need to matter the most.

Reply:
it seems to me like we're talking about two very different types of groups in terms of composition and goals, although ones that could share some common ways of working.  

1) Groups based around a craft, like PMs, Engineers, UX, etc. 
2) Groups that are cross-discipline by definition (DKAN, Sales, ?)

Reply:
Yes, DKAN sounds like a fit for the concept of a "Guild." :)

Reply:
i'm super confused @ninavizz because this doc looks a _lot_ like what we are doing in what we are currently calling the UX Practice Area. https://github.com/fmizzell/handbook/blob/e048d441eb0a5c6d5453eef512f36ba2e7b249dc/docs/02-about-us/guilds.md

Reply:
Practice Areas need to be more strategically developed... and this should be a convo offa GitHub. :)

Reply:
@ninavizz https://medium.com/yodle-tech-blog/agile-guilds-the-yodle-way-47dc00f6cd3a is the model we have been using for guilds, so in that framework they are very much _only_ cross discipline by definition. To me that you understood the opposite meaning lends more weight to what we have been feeling is that the word "guild" is way too ambiguous and abstract and "Practice Area" (for 1) is clearer for pretty much everyone.

I would suggest the naming of (2) should be (something like) a "project" (for client or goal oriented internal projects) or an "internal group" (for "department" or "product" type things like sales and DKAN).

Reply:
On an impromptu call with @kevwalsh and @fmizzell --- we're leaning toward a general concept of a "Working Group" that has categories of "subgroups" like practice areas, products, departments, etc. These terms can be flushed out in greater detail after this PR, but all "working groups" would observe some similar processes and tools like what we've proposed here.

Reply:
We want to make this as abstract as possible so that groups can self form as the need arises.

Reply:
I don't really like the idea of aiming to make this abstract - I think we need some structure to hang things off, and also because different types of groups have very different organizing principles, lifecycles and outcomes.What we have now (i.e. feel free to go form your own group) is already as abstract as it can get - I think that what we need is a (simple) support structure to make it easier to create groups and for information to flow between groups.

Reply:
@grugnog I think thats what we're aiming for. The only thing abstract now is that we're not restricting or defining the types of groups that can form (which is good, in my mind). We want to have a similar structure (that which is currently described as a guild) that can be applied to different types of groups.

Reply:
@grugnog As @nerdstein mentioned, "abstract" is only qualifying the term we are going to use for the process and artifacts. The process and artifacts are still well defined and structured, but they simply do not (or shouldn't) say anything about the purpose or content of the group, which I feel a term like "practice area" would do.

Reply:
@nerdstein OK - I am still not sure though. We have client projects and internal projects that would be very similar to department/product type working groups - multi-disciplinary, task oriented etc.

Reply:
To put it very abstractly, I feel like the distinction between a group wanting to "affect the world" is very different from a group wanting to "share skills and learn how to affect the world more effectively" in mission and that the process and artifacts needed for each are quite different.

Reply:
Client projects and internal projects are orthogonal to the "working group" (as the guild image demonstrates).I actually see them operating more like our standard projects that may potentially provide information into these working groups.

@grugnog do you have a better recommendation than "working group" to capture something more abstract than a practice area or product team that can leverage this guild-like structure?

Reply:
@grugnog I agree with that assessment. This structure will not work for all groups, only for "working groups" which speaks to some orientation toward action. That is why there is a backlog, continuity, etc. Can you think of a better term to describe the few artifacts and light process we have described so far?

Reply:
I have had a go at cleaning up the sketch from the previous time we had this discussion: https://docs.google.com/a/civicactions.net/spreadsheets/d/1mq_iUKxh4NwdA3ciGhkDqCZ2Eg519u7B5gnh35c02XI/edit?usp=sharing

Reply:
@grugnog I agree with everything in that diagram, I would say that ClientProject, InternalWorkingGroup, InternalProject, and PracticeArea can all implement the WorkingGroup interface :) The feedback loops that you are defining would probably belong in a separate Interface.

Reply:
OK - I think we are roughly on the same page in terms of the big picture. I think the discomfort I have in terms of terminology is specifically to have practice areas being a type of working group when their main orientation is around learning not working. Similarly having projects be part of this definition, even though they do work, seems like we would encounter difficulties (not all use Trello, some we can't list publically etc).

I am not sure I have a better term though, without going up to just "group" (we might just call them "thing" at that stage!).

Overall I am thinking that focusing this back to just "practice areas" would be helpful. Yes, I agree projects and internal working groups may have some of the same structures (slack channels, backlogs etc), I am not sure that presenting them as subsets is a useful distinction from the perspective of the user relative to presenting these two types of things as orthogonal.

Reply:
@grugnog @fmizzell @nerdstein @kevwalsh I guess the way I see it is that "Practice Areas" exist within a services business at the pleasure of management, as a discipline-specific component of their delivery model for revenue. Law firms will have a "Family Law" practice area and a "Criminal Law" practice area, as examples... but then within those could have "Child Custody" or "Domestic Battery" guilds. 

A practice area functions in service to a company's: 
• Marketing, via demonstrating thought leadership in a service domain;
• Brand equity, via demonstrated leadership driving innovation in a domain;
• Employees, via professional development of practitioners through mentorship, continuous learning programs, and training investments; 
• Sales and Professional Services, via collaborative expertise to shape existing and future delivery projects. 

So the long and the short of it, is that "Practice Areas" function at the pleasure of management's business strategy decisions & HR investments, whereas "Guilds" are bottom-up groups created for structured knowledge sharing. A "Practice Area" has trackable dollars behind it, revenue in front of it, and uniquely qualified 'practitioners' with larger career interests fueling their commitments to craft, driving it.

It's neither inclusive nor exclusive to holocratic org structures, so I don't think it really fits, here. It's more tied to business development, business strategy, sales strategy, marketing, and all the other much less bottom-up parts of things, yet only made possible by an army of committed actioneers. Or, Actioneers. ;)

Reply:
Supplemental analogy: HR is a "Practice Area," but "Diversity & Inclusion" would be for a Guild—but they're not interdependent, and one could easily exist w/o the other (but HR sucks w/o inclusion, maan!).

Reply:
I do not think subsets, or hierarchies, or taxonomies are the right analogy to what this initial piece of documentation is trying to get to. That is why I was trying to use OOP Interfaces as the analogy. This initial part of the documentation is not about the specifics of what a practice area is, or how projects and practice areas will interact (even though I do believe that there are some of those ideas in the PR that we should take out for now) but about the minimum set of requirements a "group" needs to be part of an ecosystem of groups. So it seems to me like we should limit this initial PR to visibility. We want people to be able to find active groups, to be able to see what they are doing, and to participate if they desire. Whatever processes and artifacts are related to that could go in first. I also know that we do have some structural suggestions to help "working groups" work. A "working group" does sound like a subset of "group", but what I am trying to get at is that, for example, a practice area could choose to be, or not to be a working group, but they would definitely need to be a group as we, at minimum, want visibility.

I guess this begs the question of "why do we need this level of abstraction if what we are trying to get going are practice areas?". Well, in DKAN we found a case in which we want to use some of the structures and artifacts that we have been talking about in a non-practice-area scenario. Only time will tell if this non-practice-area working group is ill conceived, but I believe that it does little damage to over-abstract right now, and possibly collapse these abstractions later if we find them unnecessary.

So, in case it is not clear, what I am proposing is that we focus this PR on 2 abstractions/Interfaces: Group and Working Group. Group focuses on being part of the ecosystem: Visibility. Working Group focuses on extra artifacts and processes that help a group get work done. Any other discussions can be handled in other relevant PRs. Also, I am not married to the terms group and working group, I am more interested in clarifying the ideas so we can move this PR forward with the intention of it being useful to this new DKAN group that is being formed.

Reply:
a) What is an OOP interface?
b) No interest in hierarchies—sole interest, is in only ensuring that we can build-up a UX Practice that has the correct relationship with Sales and "Management" wrt how we're pimped and utilized on projects, and financially supported for either team activities or team prodev stuff (separate from individual prodev stuff)

Yes, I'll totally cop to a lot of fear that this very dev-centric organization may inadvertently put systems into place that would/could restrict how the healthiest-possible UX Practice could structurally come together for optimal cross-org pollination, outward-facing brand equity building for recruitment and sales, and prodev support of us UX soldiers. 

Pls just know I'm acknowledging that fear, and asking that if it's seen as erroneously threatening other interests with this effort that in fact will not in any way restrict how we can gel a UX Practice, lemme know. <3

Reply:
@ninavizz In OOP (object oriented programming) an interface is a contract of interaction between 2 objects. For example, in a Customer -> Waiter interaction, there are certain thing expected from each actor. The waiter should be able to take an order, and the customer should be able to pay. Those expectations are part of the Customer or Waiter interfaces as those action are expected for those 2 actors to interact. 

What is interesting about interfaces is that even though they are prescriptive, they only care about the specific interaction and nothing else. So for example, just b/c an actor is playing the Customer role, it does not mean that that is all that actor is. The actor could also be implementing a Parent interface, or a UXer interface. I think the idea of interface is analogous to roles and accountability in the sense that one person can fulfill multiple roles in different interactions. 

As mentioned in my last comment, right now I only care about fleshing out what the minimum expectations for a group are to be visible, reachable, and accountable within the organization, and maybe putting some structure around what a working group is. If we agree on starting there, then I know that you will point out if any of the prescriptions seem to work against your idea of what a practice area is. 

But the PR as it is right now is probably not a good place to start that deeper analysis as we need to do some refactoring work as soon as we agree on where to start.

Reply:
@grugnog For the purpose of moving forward, do you think that we can concentrate on the high level idea that "we want useful groups to form to work or learn, and that we want some supportive documentation around that idea"? As you mentioned, this might be more general than is useful in practice, but I think that for the purpose of getting some useful documentation up that we can all agree upon, that might be the right level of abstraction. If this is ok, I can do some refactoring of this PR with that scope in mind, and then we can discuss further. Thoughts?